src/libraries/Microsoft.Extensions.Options/src/IOptionsFactory.cs

```diff
@@ -13,8 +14,12 @@ public interface IOptionsFactory<[DynamicallyAccessedMembers(Options.Dynamically
         where TOptions : class
     {
         /// <summary>
-        /// Returns a configured <typeparamref name="TOptions"/> instance with the given name.
+        /// Returns a configured <typeparamref name="TOptions"/> instance with the given <paramref name="name"/>.
         /// </summary>
+        /// <param name="name">The name of <typeparamref name="TOptions"/> instance to create.</param>
+        /// <returns>The created <typeparamref name="TOptions"/> instance with thw given <paramref name="name"/>.</returns>
+        /// <exception cref="OptionsValidationException">One or more <see cref="IValidateOptions{TOptions}"/> return failed <see cref="ValidateOptionsResult"/> when validating the <typeparamref name="TOptions"/> instance been created.</exception>
+        /// <exception cref="MissingMethodException">The <typeparamref name="TOptions"/> does not have a public parameterless constructor or <typeparamref name="TOptions"/> is <see langword="abstract"/>.</exception>
```

Comment:
Just wondering, do we mention the exceptions in the interface docs? I think the exception is mainly thrown from the actual interface implementation.

Reply:
That is a very good question, I don't think we should have exceptions here.

Reply:
Done, I removed the exceptions from interface docs.